pkgs/development/libraries/qt-6/qtModule.nix

```diff
@@ -32,9 +38,48 @@ stdenv.mkDerivation (args // {
         moveToOutput "$dir" "$dev"
       done
     fi
+    fixQtBuiltinPaths $out/lib "*.pr?"
     ${args.postInstall or ""}
   '';
 
+  preConfigure = ''
+    ${args.preConfigure or ""}
+
+    fixQtBuiltinPaths . '*.pr?'
+  '' + lib.optionalString (builtins.compareVersions "5.15.0" version <= 0)
+  # Note: We use ${version%%-*} to remove any tag from the end of the version
+  # string. Version tags are added by Nixpkgs maintainers and not reflected in
+  # the source version.
+  ''
+    if [[ -z "$dontCheckQtModuleVersion" ]] \
+        && grep -q '^MODULE_VERSION' .qmake.conf 2>/dev/null \
+        && ! grep -q -F "''${version%%-*}" .qmake.conf 2>/dev/null
+    then
+      echo >&2 "error: could not find version ''${version%%-*} in .qmake.conf"
+      echo >&2 "hint: check .qmake.conf and update the package version in Nixpkgs"
+      exit 1
+    fi
+
+    if [[ -z "$dontSyncQt" && -f sync.profile ]]; then
+      ${self.qtbase.dev}/libexec/syncqt.pl -version "''${version%%-*}"
```

Comment:
This should go to nativeBuildInputs

Reply:
I can add it there but it still won't be on `PATH` as it's using something from `libexec` not `bin`. 

Should it be in both places? Done that for now.

Reply:
Having it in both places makes no difference. For cross compiling it needs to be in nativeBuildInputs or use some variant of buildPackages which I am not very familiar with.

Reply:
I don't know how to fix it then :(

Reply:
I put a FIXME here since I still don't know how to fix this and having it work at all is probably more important than cross compilation.

Reply:
Fixed with help from @winterqt 
https://matrix.to/#/!kjdutkOsheZdjqYmqp:nixos.org/$pSUUcMGo7gKI_Vn7Fcdc_CkLM9rRZEIJt7uXKPLWkBM?via=nixos.org&via=matrix.org&via=nixos.dev